### DISCUSSION TITLE
Using send file option logging out consistently

### DISCUSSION THREAD

**Discussion by dr460nf1r3** [dani-garcia/vaultwarden#2753] [Q&A] [answered]
### Subject of the issue
Using the send file option triggers a logout.

### Deployment environment

* vaultwarden version: 2022.9.0 (Docker testing)
* Install method: Docker
* Clients used: Android, Firefox extension, Web
* Reverse proxy and version: Docker image macbre/nginx-http3:latest

### Steps to reproduce
1. Install the Docker `testing image` using the options listed below
2. Try to send a file via the send feature
```
DOMAIN: redacted
SIGNUPS_ALLOWED: True
SMTP_FROM: redacted
SIGNUPS_ALLOWED: True                                                                         
SMTP_FROM: redacted
SMTP_HOST: redacted
SMTP_PASSWORD: redacted
SMTP_PORT: redacted
SMTP_SSL: redacted
SMTP_USERNAME: redacted
WEBSOCKET_ENABLED: True
YUBICO_CLIENT_ID: redacted
ADMIN_TOKEN: redacted
YUBICO_SECRET_KEY: redacted
```

### Expected behaviour
Actually have the feature working without logout.

### Actual behaviour
The user gets logged out.

### Troubleshooting data
The relevant log excerpt of when the issue occurred:
```
[2022-09-18 09:19:11.266][request][INFO] POST /api/sends/file/v2
[2022-09-18 09:19:11.266][_][ERROR] No matching routes for POST /api/sends/file/v2 application/json.
[2022-09-18 09:19:11.266][_][WARN] No 404 catcher registered. Using Rocket default.
[2022-09-18 09:19:11.266][response][INFO] 404 Not Found

**Comment by BlackDex** [comment]
Could you post the support string please?

Because this shouldn't be an issue. Since it should try the v1 directly after. Also check that you do not catch proxy errors within your reverse proxy, that could also cause issues.

**Comment by dr460nf1r3** [comment]
Sorry, yes of course:

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.25.2-9c891baa
* Web-vault version: v2022.9.0
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Forwarded-For)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: false
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.35.4
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, SIGNUPS_ALLOWED, ADMIN_TOKEN, YUBICO_CLIENT_ID, YUBICO_SECRET_KEY, SMTP_HOST, SMTP_PORT, SMTP_FROM, SMTP_USERNAME, SMTP_PASSWORD


```json
{
  "_duo_akey": "***",
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*********.***********.***",
  "domain_origin": "*****://*********.***********.***",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_org_name": "Garuda's Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Forwarded-For",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": null,
  "smtp_from": "*******@***********.***",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "****.***********.***",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "off",
  "smtp_ssl": false,
  "smtp_timeout": 15,
  "smtp_username": "*******@***********.***",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": "76829",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

The Nginx error log just contains a 404 on trying to access api v2
~~~
X - - [18/Sep/2022:09:52:23 +0000] "POST /api/sends/file/v2 HTTP/2.0" 404 120 "https://URL/" "Mozilla/5.0 (Windows NT 10.0; rv:104.0) Gecko/20100101 Firefox/104.0"

**Comment by BlackDex** [comment]
Ok, either `DOMAIN` is configured incorrectly or you are accessing the admin panel using a different domain then the vault it self. But if that is not the case, then please check the domain settings. These could cause some issues.

As for the 404, that shouldn't be an issue, it should just try the v1 after that, unless you catch proxy errors with nginx, which causes the errors to be altered, and then the client API doesn't know what to do next.

Also, check the F12 developer console in the browser if you see at least an attempt, or maybe an other error there.

I just tried it my self on my system both via android client and web-vault, both work as expected

**Comment by dr460nf1r3** [comment, marked as answer]
Alright. I finally found out what the problem was after having a look at the developer console - a great idea. Cloudflare's managed ruleset decided to throw a 403 on the request. After whitelisting that particular rule for the subdomain everything works fine now.
 
![image](https://user-images.githubusercontent.com/12834713/193591987-1cc02d7e-7c31-47bf-bdc0-5db2148e2d43.png)

**Comment by GerryCrooked** [comment]
@dr460nf1r3 thank you 🐉 🔥 for your solution <3 helped me a bunch 🤟
